audioDetector.html

```diff
@@ -0,0 +1,200 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ -->
+
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>Audio detection</title>
+<link rel="stylesheet" type="text/css" href="resources/main.css">
+<link rel="icon" href="resources/favicon.ico">
+<script type="text/javascript" src="https://get.cct.ericsson.net/latest/cct.js"></script>
+<script type="text/javascript" src="exampleUtils.js"></script>
+</head>
+<body>
+
+<div class="navbar">
+    <img class="brand-line" src="resources/ericsson_brandline.jpg">
+    <div class="navbar-container">
+      <a href="/c3-web-examples">
+          <img class="navbar-logo" src="resources/ECON_116px.png">
+          <span class="navbar-title">Ericsson C3 Web examples</span>
+      </a>
+      <div id="homeServerBanner"></div>
+    </div>
+</div>
+<div class="contentContainer">
+<div class="about">
+	<h3>Audio Detection example</h3>
+	This example shows how to use an <a href="https://get.cct.ericsson.net/latest/docs/reference/AudioMeter.html">AudioMeter</a> to detect if your microphone is working correctly. It also shows how to deal with various problems that can occur when requesting microphone access.
+</div>
+<div class="view-container">
+  <button onclick="setup()">Request Microphone</button>
+  <div id="indicatorContainer">
+    <p id="indicatorText" style="color:white;"></p>
+  </div>
+</div>
+</div>
+<footer id="footer">
+  <p>
+    <span style="position: relative; left: 5vw;">&copy Ericsson 2016</span>
+    <input type=button id="toggleLog" onClick="toggleLog()" value="show log"></input>
+  </p>
+  <div id="log" class='list-group'></div>
+</footer>
+<script>
+var indicatorContainer = document.getElementById('indicatorContainer')
+var indicatorText = document.getElementById('indicatorText')
+var ERROR = 'ERROR' // Just used for clarity when calling displayMessage/2
+var WARNING = 'WARNING' // Just used for clarity when calling displayMessage/2
+
+// Constants used to represent the state (_status) of the audio monitor
+AudioMonitor.AUDIO_DETECTED = true
+AudioMonitor.AUDIO_NOT_DETECTED = false
+
+// The minimum value of what can be considered "audio detected".
+// Can (and should) be tweaked to fit the use case and environment of the AudioMonitor.
+AudioMonitor.AUDIO_DETECTION_THRESHOLD = 0.000001
+
+/**
+ * An AudioMonitor is responsible for monitoring a DeviceSource to detect whether audio is working.
+ * It emits various events depending on the status of the DeviceSource.
+ * Only one DeviceSource can be monitored at any given time.
+ * 
+ * Usage:
+ * var audioMonitor = new AudioMonitor()
+ * audioMonitor.on('event_type', callback)
+ * 
+ * audioMonitor.monitor(someDeviceSourceWithAudio)
+ */
+function AudioMonitor () {
+  cct._EmitterMixin.call(this)
```

Comment:
Should handle this by passing in some callbacks instead imo, since it's private API.